dvc/repo/experiments/__init__.py

```diff
@@ -186,8 +186,8 @@ def reproduce_celery(
         if not entries:
             return results
 
-        # TODO: re-enable --jobs concurrency
-        self.celery_queue.spawn_worker()
+        for job in range(kwargs.get("jobs", 1)):
+            self.celery_queue.spawn_worker(num=job)
```

Comment:
This should use `self.celery_queue.start_workers(jobs)` the same way we use it in `dvc queue start --jobs`

https://github.com/iterative/dvc/blob/2a888970e843fba30c1162db8a388fa3582c7780/dvc/commands/queue/start.py#L15-L17

https://github.com/iterative/dvc/blob/2a888970e843fba30c1162db8a388fa3582c7780/dvc/repo/experiments/queue/celery.py#L138

Reply:
Fixed in [78971e2](https://github.com/iterative/dvc/pull/8608/commits/78971e2b471612bba751842d6e76b4a98dc4a06b).

I also renamed `spawn_worker` to `_spawn_worker`, because it is only used here and from your comment I assume it should not be used directly.
https://github.com/iterative/dvc/blob/4c11e5b4e2a2e988a20c45dc43a0955f9fa825c5/dvc/repo/experiments/queue/celery.py#L160